### PR TITLE
chore(logging): purchases logging

### PR DESCRIPTION
**Pull request by AuHau**
Adds some basic logging for the purchases state machine following how it is done in sales.

### CODE REVIEW

File: codex/purchasing/states/cancelled.nim [hunk 1]
```diff
@@ -1,10 +1,14 @@
 import pkg/metrics
+import pkg/chronicles
 import ../statemachine
 import ./errorhandling
 import ./error
 
 declareCounter(codexPurchasesCancelled, "codex purchases cancelled")
 
+logScope:
+    topics = "marketplace purchases cancelled"
```
Comment (emizzle) [inline codex/purchasing/states/cancelled.nim:10]: Nit-pick: spacing seems off for all logScope > topics. Should be 2 spaces.

**Review by emizzle — DISMISSED** [review]
Apart from the spacing and failing tests, LGTM.

File: codex/purchasing/states/submitted.nim [hunk 2]
```diff
@@ -27,7 +33,7 @@ method run*(state: PurchaseSubmitted, machine: Machine): Future[?State] {.async.
     await subscription.unsubscribe()
 
   proc withTimeout(future: Future[void]) {.async.} =
-    let expiry = request.expiry.truncate(int64)
+    let expiry = request.expiry.truncate(int64) + 1
```
Comment (markspanbroek) [inline codex/purchasing/states/submitted.nim:36]: This is not a logging change?

**Review by markspanbroek — APPROVED** [review]
Apart from failing test, looks good. Thanks @AuHau!

File: codex/purchasing/states/submitted.nim (hunk 2 above)
Comment (AuHau) [reply codex/purchasing/states/submitted.nim:36]: Right, I stumbled upon it and changed. Will move it to the #561 as it is more related...

**Pull request merged**